Library/Formula/nsq.rb

```diff
@@ -72,6 +73,39 @@ def install
   end
 
   test do
-    system "#{bin}/nsqd", "--version"
+    begin
+      lookupd = fork do
+        system bin/"nsqlookupd"
+      end
+      sleep 2
+      d = fork do
+        system bin/"nsqd", "--lookupd-tcp-address=127.0.0.1:4160"
+      end
+      sleep 2
+      admin = fork do
+        system bin/"nsqadmin", "--lookupd-http-address=127.0.0.1:4161"
+      end
+      sleep 2
+      to_file = fork do
+        system bin/"nsq_to_file", "--topic=test", "--output-dir=#{testpath}",
+               "--lookupd-http-address=127.0.0.1:4161"
+      end
+      sleep 2
+      system "curl", "-d", "hello", "http://127.0.0.1:4151/put?topic=test"
+    ensure
+      Process.kill(9, lookupd)
+      Process.kill(9, d)
+      Process.kill(9, admin)
+      Process.kill(9, to_file)
+      Process.wait lookupd
+      Process.wait d
+      Process.wait admin
+      Process.wait to_file
+
+      assert_match "test",
+                   Pathname(`ls *.dat`.chomp).read
+      assert_match version.to_s,
+                   Pathname(`ls *.dat`.chomp).read
```

Comment:
These shouldn't be inside ensure block right. Also let do things like this `File.read(Dir["*.data"].first)`